File: solgen.portal/ClientApp/src/app/views/operating-hours/services/operating-hours.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, EventEmitter } from '@angular/core';
import { environment } from '../../../../environments/environment';
import { Observable, timer } from 'rxjs';
import { map, switchMap } from 'rxjs/operators';
import { id } from '@swimlane/ngx-datatable';
import { AbstractControl, AsyncValidatorFn } from '@angular/forms';

@Injectable({
  providedIn: 'root',
})
export class OperatingHoursService {
  public baseUri = `${environment.WebApiBaseUrl}`;
  pagedData: any;
  constructor(private http: HttpClient) {}

  getOperatingHoursList(params): Observable<any> {
    return this.http.get(`${this.baseUri}OperatingHours/GetOperatingHoursList`, {
      params: {
        filter: params.filter,
        sortColumn: params.sortColumn,
        sortDir: params.sortDir,
        page: params.page,
        pageSize: params.pageSize,
      },
    });
  }

  getOperatingHoursById(id): Observable<any> {
    return this.http.get(`${this.baseUri}OperatingHours/GetOperatingHoursById`, {
      params: {
        Id: id,
      },
    });
  }

  addOperatingHours(operatingHours): Observable<any> {
    return this.http.post(`${this.baseUri}OperatingHours/AddEditOperatingHours`, operatingHours);
  }

  deleteOperatingHours(id: string): Observable<any> {
    return this.http.post(`${this.baseUri}OperatingHours/DeleteOperatingHours`, null, {
      params: { id },
    });
  }

  deleteMultipleOperatingHours(ids: string) {
    return this.http.post(`${this.baseUri}OperatingHours/DeleteMultipleOperatingHours`, null, {
      params: { ids },
    });
  }

  deleteOperatingHoursTimeSlot(id: string): Observable<any> {
    return this.http.post(`${this.baseUri}OperatingHours/DeleteOperatingHoursTimeSlot`, null, {
      params: { id },
    });
  }

  checkDuplicateName(name, id) {
    return this.http.get(`${this.baseUri}OperatingHours/CheckOperatingHoursExist?name=${name}&id=${id}`);
  }
}

File: solgen.Service/OperatingHours/OperatingHoursService.cs
using Solgen.Core;
using Solgen.Repository.OperatingHours;
using System;
using System.Collections.Generic;
using System.Text;

namespace Solgen.Service.OperatingHours
{
    public class OperatingHoursService : IOperatingHoursService
    {
        private readonly IOperatingHoursRepository _repository;
        public OperatingHoursService(IOperatingHoursRepository repository)
        {
            _repository = repository;
        }

        public bool AddEditOperatingHours(string id, string json, string companyId, string userId)
        {
            return _repository.AddEditOperatingHours(id, json, companyId, userId);
        }

        public string CheckOperatingHoursExist(string id, string name, string companyId, string userId)
        {
            return _repository.CheckOperatingHoursExist(id, name, companyId, userId);
        }

        public string DeleteOperatingHours(string id, string companyId, string userId)
        {
            return _repository.DeleteOperatingHours(id, companyId, userId);
        }

        public string DeleteOperatingHoursTimeSlot(string id, string companyId, string userId)
        {
            return _repository.DeleteOperatingHoursTimeSlot(id, companyId, userId);
        }

        public string GetOperatingHoursById(string id, string companyId, string userId)
        {
            return _repository.GetOperatingHoursById(id, companyId, userId);
        }

        public PagedData GetOperatingHoursList(string filter, string sortColumn, string sortDir, int page, int pageSize, string companyId)
        {
            return _repository.GetOperatingHoursList(filter, sortColumn, sortDir, page, pageSize, companyId);
        }
    }
}
